### ISSUE TITLE
Add calculation of dielectronic level populations

### ISSUE THREAD

**Issue by jwreep** [wtbarnes/fiasco#269]
I don't see another issue for this, so making a new one.  [See the discussion here](https://github.com/wtbarnes/fiasco/pull/260#issuecomment-1958066757).  

Fiasco does not currently have a method to calculate level populations for a dielectronic ion.

**Comment by jwreep** [comment]
I think it's worth clarifying what is meant in this issue since I didn't describe it in particular detail.  There are no "dielectronic ions" in reality, but CHIANTI creates a separate model ion for dielectronic excitation to calculate satellite lines.  The database includes directories like "fe_17" and "fe_17d" to distinguish the two.

This is described in Section 2.2 of [the fourth CHIANTI paper](https://ui.adsabs.harvard.edu/abs/2001ApJS..134..331D/abstract):

<img width="443" alt="Screenshot 2024-12-11 at 12 35 24 PM" src="https://github.com/user-attachments/assets/725f99de-8943-4923-b551-c3a88fa58f61" />
<img width="441" alt="Screenshot 2024-12-11 at 12 35 29 PM" src="https://github.com/user-attachments/assets/9fd31555-4b6e-4194-9fe1-8b24e8b62d61" />

**Comment by wtbarnes** [comment]
I agree that this needs to be worked on, but I'm not quite sure how to go about it. Maybe this is because I don't fully understand how it is done in the IDL code. 

In v9 and above, dielectronic recombination is included in the level populations calculation through a two-ion model approach. See #254 for more details. However, prior to v9, these so-called satellite lines were treated separately by including a separate ion with a "_d" suffix in the database. This is still true for some ions even in version 9 and above. 

What I'm not sure about is how these "_d" data are actually used. In `fiasco`, if there is a "dielectronic version" of the data, it can be accessed as `_dielectronic_<dataset_name>`, e.g. `fe_15._dielectronic_elvlc` would give you the data from the file `fe/fe_15d/fe_15d.elvlc`. Is it just a matter of making additional Ion objects for these "dielectronic versions" of each ion if they exist?
